apps/dead_letter/test/unit/dead_letter_test.exs

```diff
@@ -95,6 +95,8 @@ defmodule DeadLetterTest do
     end
 
     test "returns formatted DLQ message with a reason" do
+      allow(TelemetryEvent.add_event_count(any(), [:dead_letters_handled]), return: :ok)
```

Comment:
We should have tests for these that `expect` rather than `allow` so we know that the metrics events are being created.